packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -252,46 +353,174 @@ export class OwnedAddressTCRWithAppeals extends BaseWrapper<OwnedAddressTCRWithA
       isInRevealPhase = false;
     }
 
-    // if reveal period not active
-    const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
-    const revealPeriodActive = await voting.isRevealPeriodActive(challenge);
-    if (!revealPeriodActive) {
-      isInRevealPhase = false;
+    if (pollID) {
+      if (!challenge.equals(pollID)) {
+        isInRevealPhase = false;
+      }
+    }
+
+    if (isInRevealPhase) {
+      // if reveal period not active
+      const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
+      const revealPeriodActive = await voting.isRevealPeriodActive(challenge);
+      if (!revealPeriodActive) {
+        isInRevealPhase = false;
+      }
     }
 
     return isInRevealPhase;
   }
 
+  /**
+   * Checks if a listing is currently in the Request Appeal phase
+   * @param listingAddress Address of listing to check
+   */
+  public async isInRequestAppealPhase(listingAddress: EthAddress): Promise<boolean> {
+    const appealExpiryUnixTimestamp = await this.getRequestAppealExpiryTimestamp(listingAddress);
+    const appealExpiryDate = new Date(appealExpiryUnixTimestamp.toNumber() * 1000);
+    const hasRequestedAppeal = await this.instance.getAppealRequested.callAsync(listingAddress);
+    return (!hasRequestedAppeal && appealExpiryDate > new Date());
+  }
+
+  /**
+   * Gets the expiry time of the request appeal phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getRequestAppealExpiryTimestamp(listingAddress: EthAddress): Promise<BigNumber> {
```

Comment:
Do we ever need the timestamp, or should we just return dates for easier operating